services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowValidationNegativeSpec.groovy

```diff
@@ -161,4 +166,44 @@ class FlowValidationNegativeSpec extends BaseSpecification {
         def switches = topology.getActiveSwitches()
         return [switches.first(), switches.first()]
     }
+
+    @Unroll
+    def "Unable to #action flow on a non-existent flow"() {
+        when: "Try to #action rules on a non-existent flow"
+        northbound."$method"(NON_EXISTENT_FLOW_ID)
+
+        then: "An error is received (404 code)"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 404
+        exc.responseBodyAsString.to(MessageError).errorMessage ==
+                "Could not reroute flow: Flow $NON_EXISTENT_FLOW_ID not found"
+
+        where:
+        action        | method
+        "synchronize" | "synchronizeFlow"
+        "reroute"     | "rerouteFlow"
+    }
+
+    def "Unable to get flow on a non-existent flow"() {
+        when: "Try to get rules on a non-existent flow"
```

Comment:
`"Try to get rules on a non-existent flow"` would look like this:  
`getInvolvedSwitches(flow).each{ switchId -> northbound.getSwitchRules(switchId) }`   
Here you just `Trying to get flow with non-existent flow id`